lib/runner/request-helpers-presend.js

```diff
@@ -161,21 +161,24 @@ module.exports = [
             authHandler = AuthLoader.getHandler(authType),
             authPreHook,
             authInterface,
+            logError = function (err) {
+                if (err) {
+                    run.triggers.console(context.coords,
+                        'warn',
+                        'runtime~' + authType + '.auth: could not sign the request: ' + (err.message || err));
+                }
+            },
             authSignHook = function () {
                 try {
                     authHandler.sign(authInterface, context.item.request, function (err) {
-                        // handle all types of errors in one place, see catch block
-                        if (err) { throw err; }
+                        logError(err);
```

Comment:
`done()` called on error?

Reply:
Yes @codenirvana  after **logError** **done()** will be called . Earlier we are throwing error and the try catch block only works for synchronous code . JWT token generation works async and thus handled the done inside the callback instead of throwing